### DISCUSSION TITLE
Expected behavior of `delay` with `skipAnimation`

### DISCUSSION THREAD

**Discussion by rdamborsky** [pmndrs/react-spring#1779] [Support]
Hi, I've run into a situation where my expectation is different to actual functionality. I'm wondering if this is intentional.

I have a spring animation using `delay` option. When I set `skipAnimation: true` for test suite, the delay is still applied.

While I can fix this by adding a simple condition for `delay` option, I was curious if this behavior is by design. I would expect that `delay` won't be applied if `skipAnimation` is set.

[code sandbox](https://codesandbox.io/s/react-spring-delay-74bj2)

```jsx
import { useSpring, animated, Globals } from "react-spring";

Globals.assign({
  skipAnimation: true
});

export default function App() {
  const style = useSpring({
    delay: 1500,
    from: { opacity: 0, x: 50 },
    to: { opacity: 1, x: 0 }
  });

  return (
    <div>
      hello
      <animated.div style={style}>there</animated.div>
    </div>
  );
}
```

**Comment by martyan** [comment]
Looks like there's no `G.skipAnimation` check for delayed animations. 

If this behaviour is not intentional, here's a fix for it: https://github.com/martyan/react-spring/commit/07dff1b453c8d2453082aaff3db795707bd52b88 

I can open PR if so.

**Reply by joshuaellis** [reply]
Hey @martyan, if you wanna open a PR and explain in there (so its not lost in discussions) an argument for why `skipAnimation` should ignore delay then I'm all ears. My initial reaction is you're probably right. But then thinking about the what ifs, im not so sure. But yeah, If you could submit a PR and we can discuss it, that would be great 😄
